packages/mui-base/src/ListboxUnstyled/useListbox.ts

```diff
@@ -166,6 +196,47 @@ export default function useListbox<TOption>(props: UseListboxParameters<TOption>
         event,
         props: propsWithDefaults,
       });
+
+      // Handle text navigation
+      if (event.key.length === 1) {
+        const textCriteria = textCriteriaRef.current;
```

Comment:
Why shouldn't diacritic characters strings be allowed to be used for navigating? If the option itself contains diacritic characters, user can enter diacritic characters to navigate to those option. 

Also I am not sure how to check for non-printable character, do you mean it should only allow alphabetic characters (ie a-z, A-Z)? I guess I can try using RegEx...

Reply:
> Why shouldn't diacritic characters strings be allowed to be used for navigating? 

I suppose I wasn't clear in my comment - they should be, but with the current implementation are not. In cases where they require pressing <kbd>AltGr</kbd> to enter them, two (or even three) keypresses are registered. `textCriteria.keys` could then have things like `['a', 'Control', 'AltGraph', 'a']`.

> Also I am not sure how to check for non-printable character

It could be easier to create a list of non-character keys (Alt, Control, arrow keys, etc.) on a keyboard and not run the handler if they are pressed.

Reply:
> actually, it may be enough to check if event.key.length === 1 and only allow your code to run in such case. It seems that all the special keys have longer names.

Yes, this means no changes needed in `useListbox`, just adding tests to cover this behavior is sufficient?

Reply:
Sorry, my bad, somehow I haven't noticed this is implemented in line 184 🤦‍♂️.
The test you added looks ok!